pkg/ipamd/ipamd.go

```diff
@@ -618,10 +626,6 @@ func (c *IPAMContext) StartNodeIPPoolManager() {
 }
 
 func (c *IPAMContext) updateIPPoolIfRequired(ctx context.Context) {
-	if c.enablePodENI && c.dataStore.GetTrunkENI() == "" {
```

Comment:
As discussed, let's restore the current flow as we're trading IMDS calls with EC2 calls and might contribute to API throttling issues we run in to in VPC RC.

Reply:
restored in latest revision